src/main/scala/com/datastax/gatling/plugin/metrics/HistogramLogger.scala

```diff
@@ -6,309 +6,209 @@
 
 package com.datastax.gatling.plugin.metrics
 
-import java.io.{FileOutputStream, PrintStream}
-import java.nio.file.Paths
-import java.util.Collections
-import java.util.concurrent.{ConcurrentSkipListMap, TimeUnit}
+import java.io.{Closeable, FileOutputStream, PrintStream}
+import java.nio.file.{Path, Paths}
+import java.util.concurrent.ConcurrentSkipListMap
+import java.util.concurrent.TimeUnit.{MILLISECONDS, NANOSECONDS}
 
-import com.datastax.gatling.plugin.utils.ResponseTimers
-import com.fasterxml.jackson.databind.{ObjectMapper, SerializationFeature}
-import com.fasterxml.jackson.module.scala.DefaultScalaModule
-import com.typesafe.scalalogging.LazyLogging
+import com.datastax.gatling.plugin.utils.ResponseTime
+import com.typesafe.scalalogging.StrictLogging
 import io.gatling.core.Predef._
 import io.gatling.core.session.Session
 import org.HdrHistogram._
 
-import scala.collection.JavaConverters._
-import scala.collection.mutable.ArrayBuffer
-import scala.reflect.io.File
+import scala.collection.JavaConversions._
 
-class HistogramLogger(startEpoch: Long) extends LazyLogging with MetricsLogger {
-  protected val simName: String = configuration.core.simulationClass.get.split("\\.").last
-  protected val histogramLogConfig: HistogramLogConfig = HistogramLogConfig.fromConfig()
+class HistogramLogger(startTimeMillis: Long) extends StrictLogging with MetricsLogger {
+  // FIXME When gatling is invoked from the command line, there is no simulation class
```

Comment:
Does this need to be addressed as part of this PR?

Reply:
Excellent question.  This cannot be addressed, I am afraid.

I looked at the gatling code to understand why this happens.  When we invoke gatling through `gatling.sh` in the official bundle, the gatling `Configuration` does not contain any simulation class.  Instead, a prompt allow the user to enter that class.

The problem is that [only the gatling `Configuration`, L76](https://github.com/gatling/gatling/blob/9740452061be00f823c3312eabab06b96c5e17f0/gatling-app/src/main/scala/io/gatling/app/Runner.scala#L76-L78) is exposed to the outside world (us).  The `Selection` object that is created L78 [is where the user input takes place](https://github.com/gatling/gatling/blob/9740452061be00f823c3312eabab06b96c5e17f0/gatling-app/src/main/scala/io/gatling/app/Selection.scala#L43).  And it is never used to complement the `Configuration`.

To completely fix this issue, we need yet another major refactoring.  We will have to move the HDR reporting component in `gatling-core` itself.  That way, we will have access to the user input.  Very conveniently, we will also have access to the real results folder, which will solve another issue: we will be able to save the HDR histograms in the folder they should be saved in.

There is no alternative to contributing the to gatling-core.  They have used a `sealed` trait to define all the possible reporting components.  We cannot just define a new one in the plugin and call it a day.  It has to be part of gatling-core.